### PR TITLE
Checklist: Check for first incomplete task before using it

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

* Verify there is a first incomplete task before using it.

#### Testing instructions

* Create site with a fresh WP.com user.
* Finish the checklist to 100%.
* Click "My SIte"
* Verify everything works well and WP.com doesn't error out.

cc @juliasydnor @katiebethbrown @kriskarkoski

### CODE REVIEW

**Comment by tyxla** [comment]
Appears to have been introduced in #29041.

File: client/my-sites/checklist/wpcom-checklist/component.jsx [hunk 1]
```diff
@@ -207,7 +207,7 @@ class WpcomChecklistComponent extends PureComponent {
 			completed: task.isCompleted,
 			siteSlug,
 			firstIncomplete,
-			buttonPrimary: firstIncomplete.id === task.id,
+			buttonPrimary: firstIncomplete && firstIncomplete.id === task.id,
```
Comment (spen) [inline client/my-sites/checklist/wpcom-checklist/component.jsx]: Perhaps I'm overthinking this, but I wonder if we'll ever have a situation like so:
```js
task = {
  id: undefined,
}

firstIncomplete = {
  id: undefined,
}

firstIncomplete && firstIncomplete.id === task.id // true
```

If that would be a potentiality we could do something like:
```js
const firstId = get( firstIncomplete, 'id' );
...
buttonPrimary: firstId && firstId === task.id,
```

**Comment by danhauk** [comment]
I was getting the same issue happening on the checklist page after marking off all tasks as complete. Just pushed up a change to also check for `firstIncomplete` on the `<Task/>` component as well.

**Comment by danhauk** [comment]
Thanks for the quick fix @tyxla and review @spen!

**Comment by spen** [comment]
You're very welcome!

**Comment by sgomes** [comment]
Thank you for fixing this, and apologies for introducing the bug in the first place!

**Pull request merged**